packages/validation/src/user.ts

```diff
@@ -4,60 +4,74 @@ export const USER_SOCIAL_WEBSITE = /^http(s?):\/\/\S+/i;
 export const USER_SOCIAL_RESEARCHER_ID = /^[a-z]{1,3}-\d{4}-20\d\d$/i;
 export const USER_SOCIAL_NOT_URL = /^(?!\s*http(s?):\/\/)\S+/i;
 
-const getFailedResponse = () => ({
-  isOnboardable: false,
-});
+export type UserValidationFields = Partial<
+  Pick<
+    // eslint-disable-next-line no-unused-vars
+    { [P in keyof UserResponse]: { valid: boolean } },
+    | 'questions'
+    | 'teams'
+    | 'institution'
+    | 'jobTitle'
+    | 'city'
+    | 'country'
+    | 'biography'
+    | 'skills'
+  >
+>;
 
-type UserValidationResponse = {
+export type UserValidationResponse = {
+  isOnboardable: boolean;
+} & UserValidationFields;
+
+export type UserOnboardingResult = {
+  steps?: { label: string; modalHref?: string }[];
   isOnboardable: boolean;
```

Comment:
Do we need to keep a reference to `isOnboardable`? @kmaid @jhonny-m

Reply:
Yes, the backend uses this

Reply:
That's right, on the `user.route`. Thanks:)

Reply:
That's right, on the `user.route`. Thanks:)